dockerfiles/neuroconv_latest_release_dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM python:3.11.7-slim
+LABEL org.opencontainers.image.source=https://github.com/catalystneuro/neuroconv
```

Comment:
@CodyCBakerPhD I'm finally checking out the docker files, sorry for the delay!

I run 
```
docker build -f neuroconv_latest_release_dockerfile -t neuroconv_latest_release . 
```
but had an error for `psutil`
```python-traceback
53.10 Failed to build psutil numcodecs
53.10 ERROR: Could not build wheels for psutil, numcodecs, which is required to install pyproject.toml-based projects
```
but [this](https://github.com/giampaolo/psutil/issues/2192#issuecomment-1396079848) solved it for me:

```suggestion
LABEL org.opencontainers.image.source=https://github.com/catalystneuro/neuroconv
RUN apt update && apt install musl-dev python3-dev -y
```

Reply:
I can confirm this issue on my M1 but not on anything else

Thanks for the fix, we'll see if it harms the linux or other platform builds at all